bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/handler/AbstractHomeConnectThingHandler.java

```diff
@@ -855,15 +859,41 @@ protected EventHandler defaultAmbientLightCustomColorStateEventHandler() {
         });
     }
 
+    protected EventHandler updateRemoteControlActiveAndProgramOptionsStateEventHandler() {
+        return event -> {
+            defaultBooleanEventHandler(CHANNEL_REMOTE_CONTROL_ACTIVE_STATE).handle(event);
+
+            // update available program options if update was previously delayed and remote control is enabled
+            try {
+                String programKey = programOptionsDelayedUpdate;
+                if (programKey != null && "true".equals(event.getValue())) {
```

Comment:
Could you please use `parseBoolean` (https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Boolean.html#parseBoolean(java.lang.String))

Reply:
Done